views/forms/ResetPasswordForm/index.jsx

```diff
@@ -1,62 +1,93 @@
-import Form from '../Form'
-import TextInput from '../TextInput'
 import { Button } from '@mui/material'
+import { useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+import TextInput from '../TextInput'
 
-const ResetPasswordForm = ({
-  onSubmit,
-  control,
-  errors,
-  navigate,
-  onInputChange,
-}) => {
+const schema = z
+  .object({
+    username: z.string(),
+    password: z.string().min(8),
+    confirmPassword: z.string().min(8),
+    reset_key: z.string(),
+  })
+  .superRefine(({ confirmPassword, password }, ctx) => {
+    if (confirmPassword !== password) {
+      ctx.addIssue({
+        code: 'custom',
+        message: 'Passwords do not match',
+        path: ['confirmPassword'],
+      })
+      ctx.addIssue({
+        code: 'custom',
+        message: 'Passwords do not match',
+        path: ['password'],
+      })
+    }
+  })
+
+const ResetPasswordForm = ({ initialValues, onCancel, onSubmit }) => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    defaultValues: initialValues,
+    resolver: zodResolver(schema),
+  })
   return (
-    <Form onSubmit={onSubmit}>
+    <form onSubmit={handleSubmit(onSubmit)}>
       <TextInput
         control={control}
+        errors={errors.username}
+        fullWidth
```

Comment:
is this prop required..?

Reply:
no it's not required. Are you saying it should be?

Reply:
I don't think fullWidth should be a required prop, I'm curious if that's the case now..?

Reply:
it is not a required prop